From: Dave Airlie <airlied@redhat.com>
Date: Wed, 15 Oct 2008 15:56:19 +1000
Subject: [agp] correct bug in stolen size calculations
Message-id: 1224050179.10104.9.camel@clockmaker.usersys.redhat.com
O-Subject: [RHEL5.3 BZ463853] Follow on patch to fix Cantiga/Eaglelake sizing. (NOT the 965 fix).
Bugzilla: 463853
RH-Acked-by: Brian Maly <bmaly@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Peter Martuccelli <peterm@redhat.com>

>From 144962a2c686b51778faece6e8a25b1ac066ee5f Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@dhcp-1-144.bne.redhat.com>
Date: Wed, 15 Oct 2008 15:38:32 +1000
Subject: [PATCH] intel-agp.c: correct bug in stolen size calculations.

On debugging some rawhide issues with Intel, this was uncovered.

This fix need to go into RHEL along with a corresponding fix in the -i810
X.org driver.

Signed-off-by: Dave Airlie <airlied@redhat.com>

diff --git a/drivers/char/agp/intel-agp.c b/drivers/char/agp/intel-agp.c
index ca22ca7..3e627b6 100644
--- a/drivers/char/agp/intel-agp.c
+++ b/drivers/char/agp/intel-agp.c
@@ -52,8 +52,7 @@
 		 agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_82965Q_HB || \
 		 agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_82965G_HB || \
 		 agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_82965GM_HB || \
-		 agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_82965GME_HB || \
-		 agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_IGD_HB)
+		 agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_82965GME_HB)
 
 #define IS_G33	(agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_G33_HB || \
                  agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_Q35_HB || \
@@ -61,7 +60,8 @@
 
 #define IS_G4X (agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_IGD_E_HB || \
 		agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_Q45_HB || \
-		agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_G45_HB)
+		agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_G45_HB || \
+		agp_bridge->dev->device == PCI_DEVICE_ID_INTEL_IGD_HB)
 
 /* Intel 815 register */
 #define INTEL_815_APCONT	0x51
@@ -501,7 +501,7 @@ static void intel_i830_init_gtt_entries(void)
 	} else if (IS_G4X) {
 		/* On 4 series hardware, GTT stolen is separate from graphics
 		 * stolen, ignore it in stolen gtt entries counting */
-		size = 0;
+		size = 4;
 	} else {
 		/* On previous hardware, the GTT size was just what was
 		* required to map the aperture.
